tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs

```diff
@@ -342,4 +342,36 @@ public Task IfCondition_And_InlineCondition_Test()
 
         return Verify(pipeline.Serialize());
     }
+
+    class NestedConditionedBlocks : SingleStagePipelineDefinition
+    {
+        public override string TargetFile => "nested-conditioned-blocks.yml";
+        public override SingleStagePipeline Pipeline => new()
+        {
+            Jobs =
+            {
+                new Job("job1")
+                {
+                    Steps =
+                    {
+                        If.IsBranch("main")
+                            .Step(Bash.Inline("echo 'Main branch'"))
+                            .If.IsPullRequest
+                                .StepTemplate("template1.yml")
+                            .EndIf
```

Comment:
I think the thing that does not work is this

```suggestion
                        If.IsBranch("main")
                            .If.IsPullRequest
                                .StepTemplate("template1.yml")
                            .EndIf
```

Reply:
I'll add another test case for this

Reply:
got it, this doesn't compile right now... since the `IfCondition` type doesn't have an `If` property

Reply:
Yeah, it might just need a new API but maybe it will be more complicated to make it serialize properly. But maybe it's just about creating the `Conditioned` the right way